### PR TITLE
Issue #1

### PR DESCRIPTION
**Pull request by wenta**

### CODE REVIEW

File: src/main/scala/carldata/series/TimeSeries.scala [hunk 1]
```diff
@@ -17,6 +17,34 @@ object TimeSeries {
   def empty[V: Numeric]: TimeSeries[V] = {
     new TimeSeries[V](Seq[(LocalDateTime, V)]())
   }
+
+  def resample[V: Fractional](ts: TimeSeries[V], delta: Duration)(implicit num: Fractional[V]): TimeSeries[V] = {
```
Comment (klangner) [inline src/main/scala/carldata/series/TimeSeries.scala]: Improve according to our discussion.

**Review by klangner — CHANGES_REQUESTED** [review]

File: src/main/scala/carldata/series/TimeSeries.scala [hunk 2]
```diff
@@ -16,6 +17,39 @@ object TimeSeries {
...
+
+  /**
+    * Resample given TimeSeries with indexes separated by delta
+    */
+  def resample[V: Fractional](xs: TimeSeries[V], delta: Duration)(implicit num: Fractional[V]): TimeSeries[V] = {
+    val ys: mutable.ListBuffer[V] = ListBuffer()
+    val ts = Iterator.iterate(xs.index.head)(_.plusNanos(delta.toNanos))
+      .takeWhile(_.isBefore(xs.index.last.plusNanos(1))).toVector
+
+    def g(ts: Vector[LocalDateTime], xs: Vector[LocalDateTime], vs: Vector[V], prev: V): TimeSeries[V] = {
```
Comment (klangner) [inline src/main/scala/carldata/series/TimeSeries.scala]: Add @tailrec annotation to this function

File: src/main/scala/carldata/series/TimeSeries.scala [hunk 3]
```diff
@@ -16,6 +17,39 @@ object TimeSeries {
...
+        val mu = num.plus(num.times(num.div(ty, num.plus(tx, ty)), vs.head), num.times(num.div(tx, num.plus(tx, ty)), prev))
+        ys.append(mu)
+        if (ts.size == 1) TimeSeries(ts, ys.toVector)(num) else g(ts.tail, xs, vs, mu)
+      }
+      else {
+        g(ts, xs.tail, vs.tail, vs.head)
+      }
+    }
+
+    g(ts, xs.index, xs.values, num.fromInt(0))
```
Comment (klangner) [inline src/main/scala/carldata/series/TimeSeries.scala]: The function g returns TimeSeries, but it is discarded. 
Is the next line necessary? Why?

File: src/main/scala/carldata/series/TimeSeries.scala [hunk 4]
```diff
@@ -16,6 +17,39 @@ object TimeSeries {
...
+  /**
+    * Resample given TimeSeries with indexes separated by delta
+    */
+  def resample[V: Fractional](xs: TimeSeries[V], delta: Duration)(implicit num: Fractional[V]): TimeSeries[V] = {
+    val ys: mutable.ListBuffer[V] = ListBuffer()
+    val ts = Iterator.iterate(xs.index.head)(_.plusNanos(delta.toNanos))
+      .takeWhile(_.isBefore(xs.index.last.plusNanos(1))).toVector
+
+    def g(ts: Vector[LocalDateTime], xs: Vector[LocalDateTime], vs: Vector[V], prev: V): TimeSeries[V] = {
+      val tsh = ts.head
```
Comment (klangner) [inline src/main/scala/carldata/series/TimeSeries.scala]: Try the following test:
```scala
it should "resample: empty series" in {
    val emptySeries = TimeSeries.empty[Float]
    TimeSeries.resample(emptySeries, Duration.ofMinutes(2)) shouldBe emptySeries
  }
```

**Review by klangner — CHANGES_REQUESTED** [review]

File: src/main/scala/carldata/series/TimeSeries.scala (hunk 3 above)
Comment (wenta) [reply src/main/scala/carldata/series/TimeSeries.scala]: With every recursion in 'g' ts lost previous index.  ts provided to TimeSeries is from resample scope.

File: src/main/scala/carldata/series/TimeSeries.scala (hunk 4 above)
Comment (wenta) [reply src/main/scala/carldata/series/TimeSeries.scala]: Good point. I made a silly assumption.

**Pull request merged**